syn/routes/api/v1/utils.py

```diff
@@ -70,3 +71,38 @@ def tokens():
                 res[chain][token].update({'cgid': cgid})
 
     return res
+
+
+@utils_bp.route('/token_price', methods=['GET'])
+@cache.cached()
+def token_price():
+    chain_id = request.args.get('chain_id', None, type=int)
+    token = request.args.get('token', None)
+    date = request.args.get('date', None)
+
+    # validate chain id
+    if chain_id not in CHAINS.keys():
+        return (jsonify({'error': 'invalid chain id'}), 400)
+    chain_name = CHAINS[chain_id]
```

Comment:
for required arguments we add them to the URI path, for optional arguments they are added as request params.

```suggestion
@utils_bp.route('/price/<chain:chain>/<token>', methods=['GET'])
@cache.cached()
def token_price(chain: str, token: str):
    date = request.args.get('date', type=date.fromisoformat)
    token = token.lower()
```